examples/api/lib/widgets/nested_scroll_view/nested_scroll_view.3.dart

```diff
@@ -0,0 +1,197 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [NestedScrollView].
+
+void main() => runApp(const NestedScrollViewExampleApp());
+
+class NestedScrollViewExampleApp extends StatelessWidget {
+  const NestedScrollViewExampleApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: NestedScrollViewExample(),
+    );
+  }
+}
+
+class NestedScrollViewExample extends StatefulWidget {
+  const NestedScrollViewExample({super.key});
+
+  @override
+  State<NestedScrollViewExample> createState() =>
+      NestedScrollViewExampleState();
+}
+
+class NestedScrollViewExampleState extends State<NestedScrollViewExample>
+    with TickerProviderStateMixin {
+  final List<String> _tabs = <String>['Tab 1', 'Tab 2'];
+
+  late TabController _tabController;
+  int _selectedIndex = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(
+      initialIndex: _selectedIndex,
+      vsync: this,
+      length: _tabs.length,
+    );
+    _tabController.addListener(() {
+      setState(() {
+        _selectedIndex = _tabController.index;
+      });
+    });
+  }
```

Comment:
Opportunity for a small cleanup here:

```dart
  late final TabController _tabController = TabController(
    initialIndex: _selectedIndex,
    vsync: this,
    length: _tabs.length,
  );
  int _selectedIndex = 0;

  @override
  void initState() {
    super.initState();
    _tabController.addListener(() {
      setState(() {
        _selectedIndex = _tabController.index;
      });
    });
  }
```

(It's also possible to factor out `_selectedIndex` and just reference the tab controller directly, but having this variable is perhaps the most readable way to do it.)

Reply:
Thanks! I fixed it!